packages/rspack/src/config/types.ts

```diff
@@ -1323,6 +1365,188 @@ export type ExternalItem =
 export type Externals = ExternalItem | ExternalItem[];
 //#endregion
 
+//#region ExternalsPresets
+/** Enable presets of externals for specific targets. */
+export type ExternalsPresets = {
+	/** Treat node.js built-in modules like `fs`, `path` or `vm` as external and load them via `require()` when used. */
+	node?: boolean;
+
+	/** Treat references to `http(s)://...` and `std:...` as external and load them via import when used. */
+	web?: boolean;
+
+	/** Treat references to `http(s)://...` and `std:...` as external and load them via async import() when used  */
+	webAsync?: boolean;
+
+	/** Treat common electron built-in modules in main and preload context like `electron`, `ipc` or `shell` as external and load them via `require()` when used. */
+	electron?: boolean;
+
+	/** Treat electron built-in modules in the main context like `app`, `ipc-main` or `shell` as external and load them via `require()` when used. */
+	electronMain?: boolean;
+
+	/** Treat electron built-in modules in the preload context like `web-frame`, `ipc-renderer` or `shell` as external and load them via require() when used. */
+	electronPreload?: boolean;
+
+	/** Treat electron built-in modules in the preload context like `web-frame`, `ipc-renderer` or `shell` as external and load them via require() when used. */
+	electronRenderer?: boolean;
+
+	/** Treat `NW.js` legacy `nw.gui` module as external and load it via `require()` when used. */
+	nwjs?: boolean;
+};
+
+//#endregion
+
+//#region InfrastructureLogging
+/**
+ * Represents a filter item type for infrastructure logging.
+ * Can be a RegExp, a string, or a function that takes a string and returns a boolean.
+ */
+export type FilterItemTypes = RegExp | string | ((value: string) => boolean);
+
+/**
+ * Represents filter types for infrastructure logging.
+ * Can be a single FilterItemTypes or an array of FilterItemTypes.
+ */
+export type FilterTypes = FilterItemTypes | FilterItemTypes[];
+
+/**
+ * Options for infrastructure level logging.
+ */
+export type InfrastructureLogging = {
+	/**
+	 * Append lines to the output instead of updating existing output, useful for status messages.
+	 */
+	appendOnly?: boolean;
+
+	/**
+	 * Enable colorful output for infrastructure level logging.
+	 */
+	colors?: boolean;
+
+	/**
+	 * Customize the console used for infrastructure level logging.
+	 */
+	console?: Console;
+
+	/**
+	 * Enable debug information of specified loggers such as plugins or loaders.
+	 */
+	debug?: boolean | FilterTypes;
+
+	/**
+	 * Enable infrastructure logging output.
+	 * @type {"none" | "error" | "warn" | "info" | "log" | "verbose"}
```

Comment:
Do we need to use `@type` here? it is the same as the TS type

Reply:
We don't need it.

Reply:
Updated .